src/util.c

```diff
@@ -161,6 +163,15 @@ void disconnect_mem_error(xmpp_conn_t *conn)
     xmpp_disconnect(conn);
 }
 
+int string_to_ul(const char *s, unsigned long *ul)
```

Comment:
Can be done in this way:
```C
char *endptr;
*ul = strtoul(s, &endptr, 10);
return *endptr != '\0';
```
According to man page, errno is not guaranteed on an error:
```
ERRORS
       EINVAL (not in C99) The given base contains an unsupported value.

       ERANGE The resulting value was out of range.

       The implementation may also set errno to EINVAL in case no conversion was performed (no digits seen, and 0 returned).
```